src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -129,43 +136,30 @@ private static ModelBindingContext CreateNewBindingContext(ModelBindingContext o
                     if (valueProvider == null)
                     {
                         // Unable to find a value provider for this binding source.
-                        return null;
+                        return false;
                     }
                 }
             }
 
-            var newBindingContext = new ModelBindingContext
-            {
-                Model = oldBindingContext.Model,
-                ModelMetadata = oldBindingContext.ModelMetadata,
-                FieldName = oldBindingContext.FieldName,
-                ModelState = oldBindingContext.ModelState,
-                ValueProvider = valueProvider,
-                OperationBindingContext = oldBindingContext.OperationBindingContext,
-                PropertyFilter = oldBindingContext.PropertyFilter,
-                BinderModelName = oldBindingContext.BinderModelName,
-                BindingSource = oldBindingContext.BindingSource,
-                BinderType = oldBindingContext.BinderType,
-                IsTopLevelObject = oldBindingContext.IsTopLevelObject,
-                ValidationState = oldBindingContext.ValidationState,
-            };
-
             if (bindingSource != null && bindingSource.IsGreedy)
             {
-                newBindingContext.ModelName = oldBindingContext.ModelName;
+                bindingContext.ModelName = modelName;
             }
             else if (
-                !oldBindingContext.FallbackToEmptyPrefix ||
-                newBindingContext.ValueProvider.ContainsPrefix(oldBindingContext.ModelName))
+                !fallbackToEmptyPrefix ||
+                valueProvider.ContainsPrefix(bindingContext.ModelName))
             {
-                newBindingContext.ModelName = oldBindingContext.ModelName;
+                bindingContext.ModelName = modelName;
             }
             else
             {
-                newBindingContext.ModelName = string.Empty;
+                bindingContext.ModelName = string.Empty;
             }
 
-            return newBindingContext;
+            bindingContext.ValueProvider = valueProvider;
+            bindingContext.FallbackToEmptyPrefix = false;
```

Comment:
Suggest resetting `FallbackToEmptyPrefix` for current context as part of `PushContext()`.